interfaces/builtin/allegro_vcu.go

```diff
@@ -0,0 +1,63 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2021 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+// https://www.xilinx.com/support/documentation/ip_documentation/vcu/v1_2/pg252-vcu.pdf
+// https://github.com/Xilinx/vcu-modules/
+const allegroVcuSummary = `allows access to Xilinx Allegro Video Code Unit`
+
+// Xilinx offers IP for their devices to decode/encode video streams, by
+// using /dev/allegroDecodeIP and /dev/allegroIP devices.
+// These operations should be considered privileged since the driver
+// assumes trusted input, therefore require manual connection.
+const allegroVcuBaseDeclarationSlots = `
+  allegro-vcu:
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: true
+`
+
+const allegroVcuConnectedPlugAppArmor = `
+# Description: Can access to Xilinx Allegro VCU.
+
+/dev/allegroDecodeIP rw,
+/dev/allegroIP rw,
+
+/dev/dmaproxy rw,
+`
+
+var allegroVcuConnectedPlugUDev = []string{
+	`SUBSYSTEM=="allegro_decode_class", KERNEL=="allegroDecodeIP"`,
+	`SUBSYSTEM=="allegro_encode_class", KERNEL=="allegroIP"`,
+	`SUBSYSTEM=="char", KERNEL=="dmaproxy"`,
```

Comment:
I trust these are probably correct but can you attach to the PR description or leave a comment with the udevadm output for these devices so we can verify that these rules are the "best" rules we could use to match the devices?

Reply:
```
ubuntu@ubuntu:~$ udevadm info --attribute-walk --name=/dev/allegroIP

  looking at device '/devices/virtual/allegro_encode_class/allegroIP':
    KERNEL=="allegroIP"
    SUBSYSTEM=="allegro_encode_class"
    DRIVER==""

ubuntu@ubuntu:~$ udevadm info --attribute-walk --name=/dev/allegroDecodeIP

  looking at device '/devices/virtual/allegro_decode_class/allegroDecodeIP':
    KERNEL=="allegroDecodeIP"
    SUBSYSTEM=="allegro_decode_class"
    DRIVER==""

ubuntu@ubuntu:~$ udevadm info --attribute-walk --name=/dev/dmaproxy

  looking at device '/devices/virtual/char/dmaproxy':
    KERNEL=="dmaproxy"
    SUBSYSTEM=="char"
    DRIVER==""
```

Reply:
Thanks for this.